<a href="https://colab.research.google.com/github/Nayab189/flyrank-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane


## 1. My Lane and Why

**Freestyle Capstone Title:**
Do AI-Assistant Referral Signals Improve Content Decay Prediction? A Multi-Architecture Ablation Study

**Project Rationale:**
Traditional SEO audits rely heavily on lagging Google Search Console indicators to spot declining performance. With modern search behavior increasingly influenced by conversational AI discovery (ChatGPT, Claude, Gemini, Copilot, Perplexity), a common assumption is that AI-assistant traffic serves as an immediate leading indicator for content health.

This project sets out to rigorously test that hypothesis through a multi-channel early warning system and ablation study. Rather than assuming AI traffic shifts act as active predictive drivers, this pipeline evaluates whether incorporating AI-assistant referral sessions *actually* improves the predictive accuracy of content decay models beyond what traditional search and engagement metrics achieve. By putting the AI referral hypothesis through rigorous multi-model validation (comparing architectures like HistGradientBoosting and Random Forest across cross-validation), this study provides an honest, data-driven assessment of multi-channel signals for editorial decision support.

## 2. The Question: Decision, Action, Cost of a Wrong Call

* **Research Question:** Does incorporating AI-assistant referral signals (ChatGPT, Claude, Gemini, Copilot, Perplexity) improve the predictive accuracy of a multi-channel content decay early warning system, beyond what traditional search and engagement metrics alone provide?
* **Decision:** Determine which content assets require immediate manual review and intervention versus those maintaining stable performance.
* **Who Acts:** Content editors, SEO managers, and growth strategists.
* **Unit of Analysis:** A single unique content page (`content_id`).
* **Output:** A calibrated risk probability score and a structured, tier-based priority action queue.
* **Actionable Playbook:**
  - **P1 (Full Overhaul):** Complete content restructuring and semantic rewrite for high-decay risk pages.
  - **P2 (CTR & Title Refresh):** Meta tag and title optimization for pages experiencing visibility drift.
  - **P3 / P4 (Monitor / Internal Links):** Routine monitoring or internal linking adjustments.
* **Cost of a Wrong Recommendation:**
  - *False Positive:* Misallocation of editorial resources toward updating a page that is structurally healthy.
  - *False Negative:* Failure to detect a decaying asset, resulting in an unmitigated loss of both traditional search visibility and emerging AI-assistant referral traffic.
* **Why Machine Learning:** Machine learning can model non-linear interactions among multi-channel signals that would be impractical to specify manually as fixed rules at this scale.

## 3. Quick look at the data


In [ ]:
import pandas as pd

# Loading the updated 53-column anonymized starter dataset
df = pd.read_csv("content_refresh_anonymized (1).csv")

print("Dataset Shape:", df.shape)
print("\nTrend Direction Breakdown:")
print(df["trend_direction"].value_counts())

print("\nAverage 90-Day Impressions:")
print(round(df["impressions_90d"].mean(), 2))

print("\nAverage 90-Day Sessions:")
print(round(df["sessions_90d"].mean(), 2))

Dataset Shape: (30000, 53)

Trend Direction Breakdown:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average 90-Day Impressions:
5200.37

Average 90-Day Sessions:
37.07


## Preliminary Observations (Starter Slice — Not Yet Representative)

1. **Scale & Dimensionality:** The dataset encompasses 30,000 unique content pages and 53 features, providing a sufficient sample size for initial multi-signal exploration, pending validation of representativeness against the full warehouse.
2. **Trend Distribution:** The `trend_direction` feature indicates a high concentration of declining assets (`down = 16,262`), which is consistent with content decay being a common outcome in this sample; population-level prevalence remains to be confirmed against the full warehouse in later notebooks.
3. **Traffic Viability & Unit Sanity:** Average 90-day metrics show an impression-to-session gap (~5,200 impressions vs. ~37 sessions, roughly a 0.7% conversion/interaction rate). This is consistent with long-tail skew, though the underlying cause has not yet been verified and will be examined in Notebook 2.
4. **Missing Profiling Note:** AI-assistant referral channel distributions (session volume, sparsity, and zero-inflation by assistant) have not yet been profiled here and will be prioritized in the next EDA pass given their centrality to the research question.

## 4. Careful Words: What I Can and Can't Claim

* **What this work CAN claim:**

    This system delivers **observational, directional decision-support**. It captures historical correlations between multi-channel traffic signals (including AI-assistant referrals) and subsequent content performance degradation, ranking pages to optimize human editorial workflows.
* **What this work CANNOT claim:**

    This modeling framework does **not** establish causal proof (i.e., we cannot assert that shifts in AI referrals directly *cause* search traffic drops, only that they serve as predictive indicators). Furthermore, it does **not** predict proprietary search engine algorithms or guarantee traffic recovery upon refreshing flagged assets. All outputs represent data-driven risk probabilities designed to augment, rather than replace, human editorial governance.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.